File: 6KYU/Find the missing letter/solution.cs
using System;
using System.Collections.Generic;
using System.Linq;
public class Kata
{
    public static char FindMissingLetter(char[] array)
    {
        const string alphabet = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";        
        var firstChar = char.ToUpper(array[0]);
        var startIdx = alphabet.IndexOf(firstChar);
        var missing = alphabet.ToCharArray()
                              .Skip(startIdx)
                              .Take(array.Length + 1)
                              .ToHashSet()
                              .Except(array.Select(char.ToUpper).ToHashSet())
                              .First();
        return char.IsUpper(array[0]) ? missing : char.ToLower(missing);
    }
}

File: 6KYU/Find the missing letter/solution.go
package kata

func FindMissingLetter(chars []rune) rune {
	l := len(chars) - 1
	res := rune(0)

	for i := 0; i < l; i++ {
		if chars[i+1] != chars[i]+1 {
			res = chars[i] + 1
			break
		}
	}

	return rune(res)
}
